blocks/embed/embed.js

```diff
@@ -0,0 +1,222 @@
+    })
+    .catch((error) => {
+        console.error('Error:', error);
+    });
+
+    return resultHtml.outerHTML;
+}
+
+const embedSlidShare = (url) => {
+    let resultHtml = document.createElement('div');
+    resultHtml.setAttribute('id', 'slideShare');
+    fetch(url)
+    .then(data => {
+        data.text().then( data => {
+            const $el = document.createElement('div');
+            $el.innerHTML = data;
+            const embedUrl = $el.querySelector('.slideshow-info meta[itemprop="embedURL"]').content;
+            if(embedUrl) {
+                const $slideShare = document.getElementById('slideShare')
+                $slideShare.outerHTML = `<div style="left: 0; width: 100%; height: 0; position: relative; padding-bottom: 56.25%;">
+                <iframe src="${embedUrl}" style="border:1px solid #CCC; border-width:1px; margin-bottom:5px; max-width: 100%; top: 0; left: 0; width: 100%; height: 100%; position: absolute;" frameborder="0" marginwidth="0" marginheight="0" scrolling="no" allowfullscreen loading="lazy"> </iframe>
+                </div>`;
+            }
+        });
+    })
+    .catch((error) => {
+        console.error('Error:', error);
+    });
+    
+    return resultHtml.outerHTML;
+}
+  
+const EMBEDS_CONFIG = {
+    'www.youtube.com': {
+        type: 'youtube',
+        embed: embedYoutube,
+    },
+    'video.tv.adobe.com': {
+        type: 'adobe-tv',
+        embed: getDefaultEmbed,
+    },
+    'www.instagram.com': {
+        type: 'instagram',
+        embed: embedInstagram,
+    },
+    'vimeo.com': {
+        type: 'vimeo-player',
+        embed: embedVimeo,
+    },
+    'www.vimeo.com': {
+        type: 'vimeo-player',
+        embed: embedVimeo,
+      },
+    'player.vimeo.com': {
+        type: 'vimeo-player',
+        embed: embedVimeo,
+    },
+    'spark.adobe.com': {
+        type: 'adobe-spark',
+        embed: embedSpark,
+    },
+    'twitter.com': {
+        type: 'twitter',
+        embed: embedTwitter,
+    },
+    'tiktok.com': {
+        type: 'tiktok',
+        embed: embedTiktok,
+    },
+    'www.tiktok.com': {
+        type: 'tiktok',
+        embed: embedTiktok,
+    },
+    'www.slideshare.net': {
+        type: 'slideshare',
+        embed: embedSlidShare,
+    },
+  };
+
+const loadEmbed = ($block) => {
+    const $figure = buildFigure($block.firstChild.firstChild);
+    const $a = $figure.querySelector('a');
+    if($a) {
+        const url = new URL($a.href.replace(/\/$/, ''));
+        const config = EMBEDS_CONFIG[url.hostname];
+        if (config) {
+            $a.outerHTML = config.embed(url);
+            $block.classList = `block embed embed-${config.type}`;
+        }
+        else {
+            $a.outerHTML = getDefaultEmbed(url);
+            $block.classList = `block embed embed-${getServer(url)}`;
+        }
+        $block.innerHTML = $figure.outerHTML;
+    }
+};
+
+export default function decorate($block) {
+    window.addEventListener('load', (event) => {
```

Comment:
i think this would ideally all use the `IntersectionObserver` API
https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API